src/Roslyn.Diagnostics.Analyzers/Core/DeclarePublicAPIAnalyzer.Impl.cs

```diff
@@ -359,6 +390,49 @@ internal void OnCompilationEnd(CompilationAnalysisContext context)
                 }
             }
 
+            private void ProcessTypeForwardedAttributes(Compilation compilation, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken)
+            {
+                var typeForwardedToAttribute = compilation.GetTypeByMetadataName("System.Runtime.CompilerServices.TypeForwardedToAttribute");
+
+                if (typeForwardedToAttribute != null)
+                {
+                    foreach (var attribute in compilation.Assembly.GetAttributes())
+                    {
+                        if (typeForwardedToAttribute.Equals(attribute.AttributeClass))
+                        {
+                            var forwardedType = attribute.ConstructorArguments.Single().Value as INamedTypeSymbol;
```

Comment:
This code looks fragile. There is no guarantee that there is a single argument for this attribute. Compiler matches well-known attributes by full name of the target type and signature of the constructor, no symbol comparison is involved.